src/main/java/game/map/MapHelper.java

```diff
@@ -11,9 +11,20 @@ public static Continent getContinentWithId(Map map, int id) {
                 .stream().filter(continent -> continent.getId() == id).toList().get(0);
     }
 
-    public static Country getCountryWithId(Map map, int id) {
+// Below method throws IndexOutOfBoundsException when no country is found with the given id. But the next method
+// will return "null" which seems to be a safer option. Leaving it to Shanmukh for confirmation as the initial method
+// was created by him.
 
+//    public static Country getCountryWithId(Map map, int id) {
+//
+//        return map.getCountries()
+//                .stream().filter(country -> country.getId() == id).toList().get(0);
+//    }
+    public static Country getCountryWithId(Map map, int id) {
```

Comment:
Yes, this is a better way of doing things. But people who call this method have to be aware that if they use an Id that doesn't exist, then they would get null as the response and handle it properly whenever they call the method. (whereas previously if you use a bad Id, an exception will occur and things will fail immediately).

Could you remove the code you commented and update `getContinentWithId` the same way?

Also, please check whoever used these methods handled things properly when this method return null. You can also add the code to handle the case when this method returns `null`, wherever this method is called.